data/crac/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/InjectionSetpointImplTest.java

```diff
@@ -132,6 +132,37 @@ void applyOnDanglingLine() {
         assertEquals(100., network.getDanglingLine("DL1").getP0(), 1e-3);
     }
 
+    @Test
+    void hasImpactOnNetworkForShuntCompensator() {
+        Network network = NetworkImportsUtil.import12NodesNetwork();
+        NetworkImportsUtil.addShuntCompensator(network);
+        InjectionSetpointImpl shuntCompensatorSetpoint = new InjectionSetpointImpl(
+                new NetworkElementImpl("SC1"),
+                0, Unit.SECTION_COUNT);
+        assertTrue(shuntCompensatorSetpoint.hasImpactOnNetwork(network));
+    }
+
+    @Test
+    void hasNoImpactOnNetworkForShuntCompensator() {
+        Network network = NetworkImportsUtil.import12NodesNetwork();
+        NetworkImportsUtil.addShuntCompensator(network);
+        InjectionSetpointImpl shuntCompensatorSetpoint = new InjectionSetpointImpl(
+                new NetworkElementImpl("SC1"),
+                1, Unit.SECTION_COUNT);
+        assertFalse(shuntCompensatorSetpoint.hasImpactOnNetwork(network));
+    }
+
+    @Test
+    void applyOnShuntCompensator() {
+        Network network = NetworkImportsUtil.import12NodesNetwork();
+        NetworkImportsUtil.addShuntCompensator(network);
+        InjectionSetpointImpl shuntCompensatorSetpoint = new InjectionSetpointImpl(
+                new NetworkElementImpl("SC1"),
+                1, Unit.SECTION_COUNT);
```

Comment:
it's a very bad idea to set the setpoint of the RA equal to the initial one in the network in a unit test for the apply() method. if you implement an empty apply() method, this test here would still pass. you have to set a different setpoint in order to test it correctly.

Reply:
Done !